lib/llm/src/kv_router/scheduler.rs

```diff
@@ -242,51 +242,60 @@ impl WorkerSelector for DefaultWorkerSelector {
         request: &SchedulingRequest,
         block_size: usize,
     ) -> Result<WorkerSelectionResult, KvSchedulerError> {
-        assert!(request.isl_tokens > 0);
-
-        let mut worker_scores = HashMap::new();
-        let mut max_active = 0.0;
-
-        // Calculate worker scores and find max waiting requests
-        for (worker_id, ep) in workers.endpoints.iter() {
-            // Calculate score similar to Python version
-            if let Some(score) = request.overlap.scores.get(worker_id) {
-                let score = *score as f64 * block_size as f64 / request.isl_tokens as f64;
-                worker_scores.insert(worker_id, score);
-            }
-
-            // Track max waiting requests
-            max_active = f64::max(max_active, ep.data.request_active_slots as f64);
+        // 1. Handle edge cases
+        if workers.endpoints.is_empty() {
+            // No endpoints at all
+            return Err(KvSchedulerError::NoEndpoints);
         }
 
-        if max_active == 0.0 {
+        // If block_size == 0 or isl_tokens == 0, we consider that
+        // no blocks are required. We'll still pick a worker if possible,
+        // but the overlap-based score effectively becomes 0.
+        let zero_tokens_or_blocks = (request.isl_tokens == 0) || (block_size == 0);
```

Comment:
I would assert on both of these and fail. 

One might build a case for returning an InvalidRequest for isl_tokens == 0, but block_size == 0 is a fundamental logic fail and we don't expect the system is configure correctly.  The latter should abort.